examples/operator/simple/watcher/main.go

```diff
@@ -54,20 +69,37 @@ func main() {
 		panic(fmt.Errorf("unable to add custom resource definition: %w", err))
 	}
 
-	simpleOperator, err := simple.NewOperator(simple.OperatorConfig{
-		Name:       "simple-watcher-operator",
+	// Create an operator runner for our app. This dictates how an app will be run (operator.NewRunner runs as a standalone operator)
+	runner, err := operator.NewRunner(operator.RunnerConfig{
 		KubeConfig: *kubeConfig,
-		Metrics: simple.MetricsConfig{
+		MetricsConfig: operator.RunnerMetricsConfig{
 			Enabled: true,
 		},
-		ErrorHandler: func(ctx context.Context, err error) {
-			log.Printf("\u001B[0;31mERROR: %s\u001B[0m", err.Error())
-		},
 	})
 	if err != nil {
-		panic(fmt.Errorf("unable to initialise operator: %w", err))
+		panic(fmt.Errorf("unable to create runner: %w", err))
 	}
 
+	// Set up a signal handler
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
+	defer cancel()
+
+	log.Print("\u001B[1;32mStarting Operator\u001B[0m")
+
+	// Run the app as an operator.
+	// We provide runner.Run with an AppProvider which instantiates our app using the NewApp function we define at the bottom of this file
+	err = runner.Run(ctx, simple.NewAppProvider(manifest, nil, NewApp))
+	if err != nil {
+		panic(fmt.Errorf("error running operator: %w", err))
+	}
+}
+
+type BasicModel struct {
+	Number int    `json:"numField"`
+	String string `json:"stringField"`
+}
+
+func NewApp(config app.Config) (app.App, error) {
 	// Set up the watcher
 	watcher := &simple.Watcher{
```

Comment:
@radiohead mentioned that a reconciler was preferred over a watcher. if so, should the watcher be switched for a reconciler as well?

Reply:
We have an example for each right now, I don't know if we want to remove the watcher examples completely yet (especially since codegen still builds a watcher by default at the moment)